code/modules/mob/living/living_mobility.dm

```diff
@@ -0,0 +1,154 @@
+
+//Stuff like mobility flag updates, resting updates, etc.
+
+//Force-set resting variable, without needing to resist/etc.
+/mob/living/proc/set_resting(new_resting, silent = FALSE, updating = TRUE)
+	if(new_resting != resting)
+		resting = new_resting
+		if(!silent)
+			to_chat(src, "<span class='notice'>You are now [resting? "resting" : "getting up"].</span>")
+		update_resting(updating)
+
+/mob/living/proc/update_resting(update_mobility = TRUE)
+	if(update_mobility)
+		update_mobility()
+
+//Force mob to rest, does NOT do stamina damage.
+//It's really not recommended to use this proc to give feedback, hence why silent is defaulting to true.
+/mob/living/proc/KnockToFloor(disarm_items = FALSE, silent = TRUE, updating = TRUE)
+	if(!silent && !resting)
+		to_chat(src, "<span class='warning'>You are knocked to the floor!</span>")
+	set_resting(TRUE, TRUE, updating)
+	if(disarm_items)
+		drop_all_held_items()
+
+/mob/living/proc/lay_down()
+	set name = "Rest"
+	set category = "IC"
+	if(client?.prefs?.autostand)
+		intentionalresting = !intentionalresting
+		to_chat(src, "<span class='notice'>You are now attempting to [intentionalresting ? "[!resting ? "lay down and ": ""]stay down" : "[resting ? "get up and ": ""]stay up"].</span>")
+		if(intentionalresting && !resting)
+			set_resting(TRUE, FALSE)
+		else
+			resist_a_rest()
+	else
+		if(!resting)
+			set_resting(TRUE, FALSE)
+			to_chat(src, "<span class='notice'>You are now laying down.</span>")
+		else
+			resist_a_rest()
+
+/mob/living/proc/resist_a_rest(automatic = FALSE, ignoretimer = FALSE) //Lets mobs resist out of resting. Major QOL change with combat reworks.
+	set_resting(FALSE, TRUE)
+	return TRUE
+
+//Updates canmove, lying and icons. Could perhaps do with a rename but I can't think of anything to describe it.
+//Robots, animals and brains have their own version so don't worry about them
+/mob/living/proc/update_mobility()
+	var/stat_softcrit = stat == SOFT_CRIT
+	var/stat_conscious = (stat == CONSCIOUS) || stat_softcrit
+
+	var/conscious = !IsUnconscious() && stat_conscious && !HAS_TRAIT(src, TRAIT_DEATHCOMA)
+
+	var/has_arms = get_num_arms()
+	var/has_legs = get_num_legs()
+	var/ignore_legs = get_leg_ignore()
+	var/stun = IsStun()
+	var/paralyze = IsParalyzed()
+	var/knockdown = IsKnockdown()
+	var/daze = IsDazed()
+	var/immobilize = IsImmobilized()
+
+	var/chokehold = pulledby && pulledby.grab_state >= GRAB_NECK
+	var/restrained = restrained()
+	var/pinned = resting && pulledby && pulledby.grab_state >= GRAB_AGGRESSIVE // Cit change - adds pinning for aggressive-grabbing people on the ground
+	var/canmove = !immobilize && !stun && conscious && !paralyze && (!stat_softcrit || !pulledby) && !chokehold && !IsFrozen() && (has_arms || ignore_legs || has_legs) && !pinned && !recoveringstam
+	var/canresist = !stun && conscious && !stat_softcrit && !paralyze && (has_arms || ignore_legs || has_legs) && !recoveringstam
+
+	if(canmove)
+		mobility_flags |= MOBILITY_MOVE
```

Comment:
use the bitflag update macros here

Reply:
obviously "here" means "this whole dang file"

Reply:
seconding this